src/components/header-text-with-dropdown/HeaderTextWithDropdown.tsx

```diff
@@ -0,0 +1,120 @@
+import Box from '@mui/material/Box'
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp'
+import IconButton from '@mui/material/IconButton'
+import MoreVertIcon from '@mui/icons-material/MoreVert'
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown'
+import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline'
+
+import AppTextField from '~/components/app-text-field/AppTextField'
+import { menuTypes } from '~/containers/course-section/CourseSectionContainer.constants'
+import {
+  TextFieldVariantEnum,
+  ColorEnum,
+  CourseSection,
+  FormInputValueChange
+} from '~/types'
+import { MenuItem } from '@mui/material'
+import { useTranslation } from 'react-i18next'
+import { Dispatch, SetStateAction, useState } from 'react'
+import useMenu from '~/hooks/use-menu'
+import { styles } from '~/components/header-text-with-dropdown/HeaderTextWithDropdown.styles'
+
+interface HeaderTextWithDropdownProps {
+  sectionData: CourseSection
+  setIsVisible: Dispatch<SetStateAction<boolean>>
+  onDeleteSection?: () => void
+  handleSectionInputChange?: FormInputValueChange<string, CourseSection>
+  isVisible: boolean
+  isView?: boolean
+}
+
+const HeaderTextWithDropdown = ({
+  setIsVisible,
+  handleSectionInputChange,
+  sectionData,
+  onDeleteSection,
+  isVisible,
+  isView
+}: HeaderTextWithDropdownProps) => {
+  const { t } = useTranslation()
+  const [titleInput, setTitleInput] = useState<string>(sectionData.title)
+  const [activeMenu, setActiveMenu] = useState<string>('')
+  const { openMenu, renderMenu } = useMenu()
+
+  const onShowHide = () => {
+    setIsVisible((isVisible) => !isVisible)
+  }
+
+  const sectionActions = [
+    {
+      id: 1,
+      label: (
+        <Box sx={styles.deleteIconWrapper}>
+          <DeleteOutlineIcon sx={styles.menuIcon} />
+          {t('course.courseSection.sectionMenu.deleteSection')}
+        </Box>
+      ),
+      func: onDeleteSection
+    }
+  ]
+
+  const sectionMenuItems = sectionActions.map(({ label, func, id }) => (
```

Comment:
I suggest refactoring `func` to `onClick` to make it more readable

Reply:
fixed